src/overlays/actors/ovl_En_Slime/z_en_slime.c

```diff
@@ -45,146 +77,1170 @@ ActorInit En_Slime_InitVars = {
+    } else {
+        this->distLimit = this->actor.shape.rot.x * 40.0f;
+    }
+
+    this->actor.shape.rot.x = 0;
+    reviveTimeSeconds = EN_SLIME_GET_REVIVE_TIME(&this->actor);
+    this->actor.shape.rot.z = 0;
+    this->actor.params &= 0xFF;
+    if (reviveTimeSeconds == 0xFF) {
+        reviveTimeSeconds = 0;
+    }
+
+    // Upper byte of actorParams at init is # of seconds past 10s to revive.
+    this->reviveTime = (reviveTimeSeconds * 20) + 200;
+
+    // Update addresses of texture assets (if another instance hasn't already)
+    if (!sTexturesDesegmented) {
+        for (i = 0; i < EN_SLIME_EYETEX_MAX; i++) {
+            sEyeTextures[i] = Lib_SegmentedToVirtual(sEyeTextures[i]);
+        }
+        sSlimeTexAnim = Lib_SegmentedToVirtual(gChuchuSlimeFlowTexAnim);
+        sTexturesDesegmented = true;
+    }
+
+    // Set hint IDs and drop item icon depending on chuchu color
+    if (EN_SLIME_GET_TYPE(&this->actor) == EN_SLIME_TYPE_YELLOW) {
+        this->itemDropTex = Lib_SegmentedToVirtual(gDropArrows1Tex);
+        this->actor.hintId = TATL_HINT_ID_YELLOW_CHUCHU;
+    } else if (EN_SLIME_GET_TYPE(&this->actor) == EN_SLIME_TYPE_GREEN) {
+        this->itemDropTex = Lib_SegmentedToVirtual(gDropMagicSmallTex);
+        this->actor.hintId = TATL_HINT_ID_GREEN_CHUCHU;
+    } else if (EN_SLIME_GET_TYPE(&this->actor) == EN_SLIME_TYPE_RED) {
+        this->itemDropTex = Lib_SegmentedToVirtual(gDropRecoveryHeartTex);
+        this->actor.hintId = TATL_HINT_ID_RED_CHUCHU;
+    } else {
+        this->actor.hintId = TATL_HINT_ID_BLUE_CHUCHU;
+    }
+
+    // Sets actor scale and initial action func
+    EnSlime_SetupInitializeIdle(this);
+}
+
+void EnSlime_Destroy(Actor* thisx, PlayState* play) {
+    EnSlime* this = THIS;
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+/**
+ * Upon immobilization via ice arrow (for chuchu types outside of blue),
+ * initialize ice visual effects before chuchu enters immobile state.
+ */
+void EnSlime_Freeze(EnSlime* this) {
+    this->drawDmgEffType = ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX;
+    this->drawDmgEffScale = 0.4f;
+    this->collider.base.colType = COLTYPE_HIT3;
+    this->drawDmgEffFrozenSteamScale = 0.6f;
+    this->drawDmgEffAlpha = 1.0f;
+    this->timer = 80;
+    this->actor.flags &= ~ACTOR_FLAG_400;
+    Actor_SetColorFilter(&this->actor, COLORFILTER_COLORFLAG_RED, 255, COLORFILTER_BUFFLAG_XLU, 80);
+}
+
+/**
+ * If stunned ice effect is active (ie. non-blue chuchu was frozen by ice arrow),
+ * initialize thaw and spawn thawing effect.
+ */
+void EnSlime_Thaw(EnSlime* this, PlayState* play) {
+    if (this->drawDmgEffType == ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX) {
+        this->drawDmgEffType = 0; // So it's not triggered again until Freeze has been called again.
+        this->collider.base.colType = COLTYPE_NONE;
+        this->drawDmgEffAlpha = 0.0f;
+        Actor_SpawnIceEffects(play, &this->actor, this->limbPos, EN_SLIME_LIMBPOS_COUNT, 2, 0.2f, 0.2f);
```

Comment:
I know these are the same number, but this is "more correct":
```suggestion
        Actor_SpawnIceEffects(play, &this->actor, this->limbPos, ARRAY_COUNT(this->limbPos), 2, 0.2f, 0.2f);
```